src/compiler/commandLineParser.ts

```diff
@@ -462,7 +462,6 @@ namespace ts {
         {
             name: "noEmit",
             type: "boolean",
-            affectsEmit: true,

```

Comment:
Personally, I find this surprising enough that I'd leave it present but commented out and with an explanation.